File: Dev/unitTest_Engine_cs/Graphics/3D/DrawSpriteAdditionally3D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics._3D
{
	class DrawSpriteAdditionally3D : EngineGraphics3DTest
	{
		public DrawSpriteAdditionally3D()
			: base(10)
		{
		}

		protected override void OnStart()
		{
			base.OnStart();
		}

		protected override void OnUpdating()
		{
			base.OnUpdating();

			Layer3D.DrawSpriteAdditionally(
			new asd.Vector3DF(-1, 0, -1),
			new asd.Vector3DF(1, 0, -1),
			new asd.Vector3DF(1, 0, 1),
			new asd.Vector3DF(-1, 0, 1),
			new asd.Color(255, 0, 0, 255),
			new asd.Color(0, 255, 0, 255),
			new asd.Color(0, 0, 255, 255),
			new asd.Color(255, 255, 255, 255),
			new asd.Vector2DF(0, 0),
			new asd.Vector2DF(0, 0),
			new asd.Vector2DF(0, 0),
			new asd.Vector2DF(0, 0),
			null,
			asd.AlphaBlendMode.Blend,
			false,
			true
			);
		}
	}
}

File: Dev/unitTest_Engine_cpp_gtest/Graphics/3D/DrawSpriteAdditionally3D.cpp


#include "../../EngineGraphics3DTest.h"

class Graphics_DrawSpriteAdditionally3D : public EngineGraphics3DTest
{
public:

	Graphics_DrawSpriteAdditionally3D(bool isOpenGLMode) :
		EngineGraphics3DTest(asd::ToAString("DrawSpriteAdditionally3D"), isOpenGLMode, 10, true)
	{}

protected:
	void OnStart() override
	{
		EngineGraphics3DTest::OnStart();	
	}

	void OnUpdating() override
	{
		EngineGraphics3DTest::OnUpdating();

		GetLayer3D()->DrawSpriteAdditionally(
			asd::Vector3DF(-1, 0, -1),
			asd::Vector3DF(1, 0, -1),
			asd::Vector3DF(1, 0, 1),
			asd::Vector3DF(-1, 0, 1),
			asd::Color(255, 0, 0, 255),
			asd::Color(0, 255, 0, 255),
			asd::Color(0, 0, 255, 255),
			asd::Color(255, 255, 255, 255),
			asd::Vector2DF(0, 0),
			asd::Vector2DF(0, 0),
			asd::Vector2DF(0, 0),
			asd::Vector2DF(0, 0),
			nullptr,
			asd::AlphaBlendMode::Blend,
			false,
			true
			);
	}
};

ENGINE_TEST(Graphics, DrawSpriteAdditionally3D)